src/lib/Parser.hs

```diff
@@ -285,6 +335,41 @@ decl = do
   rhs <- sym "=" >> blockOrExpr
   return $ lhs rhs
 
+interfaceInstance :: Parser UDecl
+interfaceInstance = do
+  keyWord InstanceKW
+  (p, pos) <- withPos letPat
+  ann <- annot uType
+  keyWord WhereKW
+  rhs <- mkRhs ann <$> (withSrc $ URecord <$> interfaceRecordFields "=")
+  let (ann', rhs') = addImplicitImplicitArgs pos (Just ann) rhs
+  return $ ULet InstanceLet (p, ann') rhs'
+  where
+    -- Here, we are traversing the type annotation to retrieve the name of
+    -- the interface and generate its corresponding constructor.
+    mkRhs (WithSrc _ (UPi _ _ lrhs))      rhs = mkRhs lrhs rhs
```

Comment:
nit: could you use a more descriptive name than `lrhs`? Also `mkRhs` is a bit non-descriptive too.

Reply:
Also, maybe we should check to make sure it's an implicit arrow? Since we shouldn't really be traversing non-implicit arrows here.